pydicom/waveforms/numpy_handler.py

```diff
@@ -164,10 +163,94 @@ def generate_multiplex(ds: "Dataset", as_raw: bool = True) -> Generator["np.ndar
         if not as_raw:
             # Apply correction factor (if possible)
             arr = arr.astype('float')
-            for jj, channel in enumerate(item.ChannelDefinitionSequence):
-                if "ChannelSensitivityCorrectionFactor" not in channel:
-                    continue
-
-                arr[..., jj] *= channel.ChannelSensitivityCorrectionFactor
+            for jj, ch in enumerate(item.ChannelDefinitionSequence):
+                baseline = ch.get("ChannelBaseline", 0.0)
+                sensitivity = ch.get("ChannelSensitivity", 1.0)
+                correction = ch.get("ChannelSensitivityCorrectionFactor", 1.0)
+                arr[..., jj] = (
+                    (arr[..., jj] + baseline) * sensitivity * correction
+                )
 
         yield arr
+
+
+def multiplex_array(ds: "Dataset", index: int, as_raw: bool = True) -> "np.ndarray":  # noqa
+    """Return an :class:`~numpy.ndarray` for the multiplex group in the
+    *Waveform Sequence* at `index`.
+
+    .. versionadded:: 2.1
+
+    Parameters
+    ----------
+    ds : pydicom.dataset.Dataset
+        The :class:`Dataset` containing a :dcm:`Waveform
+        <part03/sect_C.10.9.html>` module and the *Waveform Sequence* to be
+        converted.
+    index : int
+        The index of the multiplex group to return.
+    as_raw : bool, optional
+        If ``True`` (default), then return the raw unitless waveform data. If
+        ``False`` then attempt to convert the raw data for each channel to the
+        quantity specified by the corresponding (003A,0210) *Channel
+        Sensitivity* unit.
+
+    Returns
+    -------
+    np.ndarray
+        The waveform data for a multiplex group as an :class:`~numpy.ndarray`
+        with shape (samples, channels).
+    """
+    transfer_syntax = ds.file_meta.TransferSyntaxUID
+    # The check of transfer syntax must be first
+    if transfer_syntax not in SUPPORTED_TRANSFER_SYNTAXES:
+        raise NotImplementedError(
+            "Unable to convert the waveform data as the transfer syntax "
+            "is not supported by the waveform data handler"
+        )
+
+    if 'WaveformSequence' not in ds:
+        raise AttributeError(
+            "No (5400,0100) Waveform Sequence element found in the dataset"
+        )
+
+    item = ds.WaveformSequence[index]
```

Comment:
What if that doesn't exist at that index? I guess relying on the default exception is ok here... but maybe add a test for this case.

Reply:
It'll just be the standard `IndexError`, which should be pretty familiar to Python users